pkg/controller/petset/pet.go

```diff
@@ -251,8 +251,8 @@ func (p *apiServerPetClient) SyncPVCs(pet *pcb) error {
 		if !exists {
 			if err := p.createPVC(&pet.pvcs[i]); err != nil {
 				errMsg += fmt.Sprintf("Failed to create %v: %v", pvc.Name, err)
+				p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
 			}
-			p.event(pet.parent, "Create", fmt.Sprintf("pvc: %v", pvc.Name), err)
```

Comment:
Oh! okay, I understand. Then the return values from `getPVC(..)` could be, like you said:

If found: `exists=true, err=nil`
If err && (err == NotFound): `exists=false, err=nil` (because it isn't necessarily an error, unless we fail to create the PVC in the next step.)
If err : `exists=false, err=err` (this should be an event?)

Within `SyncPVCs(..)`:
successful/failed creation of a PVC would trigger events.